src/app/ui-components/input-date/input-date.component.ts

```diff
@@ -0,0 +1,100 @@
+import { Component, forwardRef, Injector, Input, OnDestroy, OnInit } from '@angular/core';
+import {
+  AbstractControl,
+  ControlValueAccessor,
+  FormControl,
+  FormControlName,
+  FormGroupDirective,
+  FormsModule,
+  NG_VALIDATORS,
+  NG_VALUE_ACCESSOR,
+  NgControl,
+  ReactiveFormsModule,
+  ValidationErrors
+} from '@angular/forms';
+import { InputMaskModule } from 'primeng/inputmask';
+import { DatePipe } from '@angular/common';
+import { FormErrorComponent } from '../form-error/form-error.component';
+import { convertDateToString, convertStringToDate } from 'src/app/utils/input-date';
+
+@Component({
+  selector: 'alg-input-date',
+  templateUrl: './input-date.component.html',
+  styleUrls: [ './input-date.component.scss' ],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => InputDateComponent),
+      multi: true,
+    },
+    {
+      provide: NG_VALIDATORS,
+      useExisting: forwardRef(() => InputDateComponent),
+      multi: true,
+    },
+  ],
+  standalone: true,
+  imports: [
+    InputMaskModule,
+    ReactiveFormsModule,
+    FormsModule,
+    DatePipe,
+    FormErrorComponent,
+  ],
+})
+export class InputDateComponent implements OnInit, OnDestroy, ControlValueAccessor {
+  @Input() minDate?: Date;
+
+  input = '';
+  control?: FormControl<Date | null>;
+
+  constructor(private injector: Injector) {
+  }
+
+  ngOnInit(): void {
+    const injectedControl = this.injector.get(NgControl);
+    if (injectedControl instanceof FormControlName) {
+      this.control = this.injector.get(FormGroupDirective).getControl(injectedControl);
+    }
+  }
+
+  ngOnDestroy(): void {
+    setTimeout(() => {
+      this.control?.clearValidators();
+      this.control?.updateValueAndValidity();
+      this.control?.patchValue(null);
+    });
```

Comment:
For my point when the control component has removed from parent form component - it should remove value and validation (primeng didn't that), otherwise it need to be handled in parent component which makes reusing this component complicated

But without timeout it doesn't work + it throws `ExpressionChangedAfterItHasBeenCheckedError` error